test/object-store/c_api/c_api.cpp

```diff
@@ -449,6 +449,99 @@ TEST_CASE("C API (non-database)", "[c_api]") {
             CHECK(std::string{realm_config_get_fifo_path(config.get())} == "test_path.FIFO");
         }
     }
+
+#if REALM_ENABLE_SYNC
+    SECTION("realm_app_config_t") {
+        // realm_http_transport is not defined if REALM_ENABLE_SYNC is not set
+        auto http_transport = realm_http_transport(nullptr);
+        auto app_config = cptr(realm_app_config_new("some_app_id", &http_transport));
+
+#if REALM_ENABLE_AUTH_TESTS
+        SECTION("realm_app_config_t with transport") {
+            std::shared_ptr<app::GenericNetworkTransport> transport = std::make_shared<SynchronousTestTransport>();
+            auto http_transport = realm_http_transport(transport);
+            app_config = cptr(realm_app_config_new("app_id_123", &http_transport));
+            CHECK(app_config.get() != nullptr);
+            CHECK(app_config->app_id == "app_id_123");
+            CHECK(app_config->transport == transport);
+        }
+#else
+        // SynchronousTestTransport is not available if SYNC is not enabled
+        SECTION("realm_app_config_t with transport") {
+            auto http_transport = realm_http_transport(nullptr);
+            app_config = cptr(realm_app_config_new("app_id_123", &http_transport));
+            CHECK(app_config.get() != nullptr);
+            CHECK(app_config->app_id == "app_id_123");
+            CHECK(app_config->transport == nullptr);
+        }
+#endif // REALM_ENABLE_AUTH_TESTS
+
+        SECTION("realm_app_config_set_base_url()") {
+            realm_app_config_set_base_url(app_config.get(), "https://path/to/app");
+            CHECK(app_config->base_url == "https://path/to/app");
+        }
+
+        SECTION("realm_app_config_set_local_app_name()") {
```

Comment:
I don't think these various calls need to be separate SECTION's. For each SECTION, the SECTION or TEST_CASE that surrounds it gets re-run. So we're basically calling the whole `SECTION("realm_app_config_t")` for each one of these. If they had side effects where calling realm_app_config_set_default_request_timeout() effected the behavior of realm_app_config_set_platform() then having a bunch of different sections would def be appropriate. But as it is, this seems like overkill.

Reply:
Merged into one section